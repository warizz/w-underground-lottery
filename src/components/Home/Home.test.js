import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Uut from './';

describe('snapshot testing', () => {
  it('should match snapshot when no props', () => {
    const props = {
      currentPeriod: {
        endedAt: new Date(2017, 0, 1),
      },
    };
    const wrapper = shallow(<Uut {...props} />);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot when props.currentPeriod.isOpen', () => {
    const props = {
      currentPeriod: {
        endedAt: new Date(2017, 0, 1),
        isOpen: true,
      },
    };
    const wrapper = shallow(<Uut {...props} />);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot when props.currentPeriod.result', () => {
    const props = {
      currentPeriod: {
        endedAt: new Date(2017, 1, 1),
        result: {},
      },
    };
    const wrapper = shallow(<Uut {...props} />);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
