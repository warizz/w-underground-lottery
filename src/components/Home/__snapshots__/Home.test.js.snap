// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing should match snapshot when no props 1`] = `
<div
  className="Home"
>
  <div
    className="pane visible-md visible-lg"
  >
    <UserProfile
      onLogOut={[Function]}
      user={Object {}}
    />
  </div>
  <div
    className="pane center"
  >
    <div
      className="visible-xs visible-sm"
      style={
        Object {
          "display": "flex",
          "margin": "0 0 8px 0",
        }
      }
    >
      <UserProfile
        onLogOut={[Function]}
        user={Object {}}
      />
    </div>
    <div
      className="visible-xs"
      id="bet-editor-container"
      style={
        Object {
          "display": "flex",
        }
      }
    />
    <BetList
      bets={Array []}
      calculateTicketPrice={[Function]}
      deleteHandler={[Function]}
      editHandler={[Function]}
      isEditable={false}
      periodEndedAt="05 Dec 2017"
    />
  </div>
  <div
    className="hidden-xs"
  >
    <div
      className="hidden-xs"
      id="bet-editor-container"
      style={
        Object {
          "display": "flex",
        }
      }
    />
  </div>
</div>
`;

exports[`snapshot testing should match snapshot when props.currentPeriod.isOpen 1`] = `
<div
  className="Home"
>
  <div
    className="pane visible-md visible-lg"
  >
    <UserProfile
      onLogOut={[Function]}
      user={Object {}}
    />
  </div>
  <div
    className="pane center"
  >
    <div
      className="visible-xs visible-sm"
      style={
        Object {
          "display": "flex",
          "margin": "0 0 8px 0",
        }
      }
    >
      <UserProfile
        onLogOut={[Function]}
        user={Object {}}
      />
    </div>
    <div
      className="visible-xs"
      id="bet-editor-container"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <BetEditor
        editingBet={Object {}}
        onClose={[Function]}
        saveBetHandler={[Function]}
      />
    </div>
    <BetList
      bets={Array []}
      calculateTicketPrice={[Function]}
      deleteHandler={[Function]}
      editHandler={[Function]}
      isEditable={true}
      periodEndedAt="05 Dec 2017"
    />
  </div>
  <div
    className="hidden-xs"
  >
    <div
      className="hidden-xs"
      id="bet-editor-container"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <BetEditor
        editingBet={Object {}}
        onClose={[Function]}
        saveBetHandler={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`snapshot testing should match snapshot when props.currentPeriod.result 1`] = `
<div
  className="Home"
>
  <div
    className="pane visible-md visible-lg"
  >
    <UserProfile
      onLogOut={[Function]}
      user={Object {}}
    />
  </div>
  <div
    className="pane center"
  >
    <div
      className="visible-xs visible-sm"
      style={
        Object {
          "display": "flex",
          "margin": "0 0 8px 0",
        }
      }
    >
      <UserProfile
        onLogOut={[Function]}
        user={Object {}}
      />
    </div>
    <div
      className="visible-xs"
      id="bet-editor-container"
      style={
        Object {
          "display": "flex",
        }
      }
    />
    <div
      style={
        Object {
          "margin": "0 0 8px 0",
        }
      }
    >
      <Reward
        bets={Array []}
        result={Object {}}
      />
      <ResultDisplay
        firstThree="-"
        fourthThree="-"
        secondThree="-"
        six="-"
        thirdThree="-"
        two="-"
      />
    </div>
    <BetList
      bets={Array []}
      calculateTicketPrice={[Function]}
      deleteHandler={[Function]}
      editHandler={[Function]}
      isEditable={false}
      periodEndedAt="05 Dec 2017"
    />
  </div>
  <div
    className="hidden-xs"
  >
    <div
      className="hidden-xs"
      id="bet-editor-container"
      style={
        Object {
          "display": "flex",
        }
      }
    />
  </div>
</div>
`;
